src/client/api/init.c

```diff
@@ -143,6 +146,23 @@ daos_init(void)
 	if (rc != 0)
 		D_GOTO(unlock, rc);
 
+	d_fault_init = d_fault_attr_lookup(101);
+
+	/* If fault injection 101 is set then turn off fault injection 0 for the rest of this
+	 * function.  This allows us to only test daos_init() under fault injection for one
+	 * test only, then avoid replicating the same effort for other fault injection tests.
+	 */
```

Comment:
This feels like a bit of a kludge to me. Would it make more sense to keep code 0 for all non-`daos_init()` cases, and have a special code for fault injection in `daos_init()`?

Reply:
0 is used for D_ALLOC() and friends, when the macro is used we simply don't know if it's inside, or outside daos_init(), in fact for the hash tables for example it might be both.

The issue here is that of performance, there are a **lot** of allocations in daos_init(), so this change has a noticeable effect on time-to-result in CI.